### PR TITLE
br: support snapshot + log restore and support different tasks for checkpoint

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: ref #43133 

Problem Summary:
make an enhancement for checkpoint for restore, this PR supports the first 2 items.
### What is changed and how it works?
1. This PR add the `taskInfo` marker to mark what the progress of the task is and what the last restore task is.
2. By the marker, we can fix the issue: `snapshot restore` create tables that renamed by `log restore`, and `log restore` rename the tables again, results in duplicate tables.
3. Use the `<start-ts, restore-ts>` to mark a task, which is in `taskInfo`. So we can know whether the current task is the last one.

NOTE: This PR also adds the `rewriteTS` into taskInfo. This is an placeholder to support the third item in issue #43133 in the future.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/client.go [hunk 1]
```diff
@@ -2340,15 +2354,70 @@ func (rc *Client) CleanUpKVFiles(
...
 // InitSchemasReplaceForDDL gets schemas information Mapping from old schemas to new schemas.
 // It is used to rewrite meta kv-event.
 func (rc *Client) InitSchemasReplaceForDDL(
-	dbMaps []*backuppb.PitrDBMap,
+	ctx context.Context,
+	tiflashRecorder *tiflashrec.TiFlashRecorder,
 	tables *map[int64]*metautil.Table,
 	tableFilter filter.Filter,
+	newTask bool,
+	hasFullRestore bool,
```
Comment (YuJuncen) [inline br/pkg/restore/client.go:2391]: if there are optional or default arguments, can we make them a config? So when are going to add more arguments in the future, we just need to add new fields in the config, and no need to change every call site of this function.

```go
type InitSchemaConfig struct {
  IsNewTask bool
  HasFullRestore bool
  TiFlashRecorder *tiflashrec.TiFlashRecorder
}
```

**Review by YuJuncen — COMMENTED** [review]
Rest LGTM

**Review by YuJuncen — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**